packages/angular/cli/upgrade/version.ts

```diff
@@ -61,6 +68,26 @@ export class Version {
       process.exit(2);
     }
 
+    try {
+      const angularCliPkgPath = require.resolve('@angular/cli/package.json', resolveOptions);
+      angularCliPkgJson = require(angularCliPkgPath);
+      if (!(angularCliPkgJson && angularCliPkgJson['version'])) {
+        throw new Error();
+      }
+    } catch (error) {
+      console.error(terminal.bold(terminal.red(tags.stripIndents`
+        Cannot determine versions of "@angular/cli".
+        This likely means your local installation is broken. Please reinstall your packages.
+      `)));
+      process.exit(2);
+    }
+
+    const angularCliVersion = new Version(angularCliPkgJson['version']);
+    const upperBound = angularCliVersion.major;
+    const lowerBound = upperBound - 1;
+    // e.g. CLI 8.0 supports '>=7.0.0 <9.0.0', including pre-releases (betas, rcs, snapshots).
+    const supportedAngularSemver = `>=${lowerBound}.0.0 <${upperBound + 1}.0.0`;
```

Comment:
@alexeagle can you verify if the N and N-1 versions are the right supported range?

Reply:
It seems right to me. let me check with Stephen to validate...